server/service/service.go

```diff
@@ -269,29 +284,121 @@ func (s *Service) ReportStatus(ctx context.Context, req *bpb.ReportStatusRequest
+			}
+			bootstrapReq := req.BootstrapRequest
+			identity := bootstrapReq.GetIdentity()
+			log.Infof("Received initial BootstrapRequest: %+v", bootstrapReq)
+
+			if identity == nil {
+				return status.Errorf(codes.InvalidArgument, "identity field is missing in BootstrapRequest")
+			}
+
+			lu, err := buildEntityLookup(ctx, bootstrapReq)
+			if err != nil {
+				return status.Errorf(codes.InvalidArgument, "failed to build entity lookup from request: %v", err)
+			}
+			chassis, err := s.em.ResolveChassis(ctx, lu, bootstrapReq.GetControlCardState().GetSerialNumber())
+			if err != nil {
+				return status.Errorf(codes.NotFound, "failed to resolve chassis: %v", err)
+			}
+			session.chassis = chassis
+			session.activeControlCard = bootstrapReq.GetControlCardState().GetSerialNumber()
+
+			// Set deviceID for logging and potential future use in the session.
+			if lu.SerialNumber != "" {
+				deviceID = lu.SerialNumber
+			} else {
+				deviceID = chassis.Serial
+			}
+			if deviceID == "" {
+				return status.Errorf(codes.InvalidArgument, "unable to determine device unique identifier")
+			}
+			log.Infof("Resolved device: %s, Chassis: %s, Active CC: %s", deviceID, session.chassis.Hostname, session.activeControlCard)
+
+			switch idType := identity.Type.(type) {
+			case *bpb.Identity_IdevidCert:
+				log.Infof("Detected IDevID flow for %s", deviceID)
+				certDERR, err := base64.StdEncoding.DecodeString(idType.IdevidCert)
+				if err != nil {
+					return status.Errorf(codes.InvalidArgument, "failed to decode idevid_cert: %v", err)
+				}
+				cert, err := x509.ParseCertificate(certDERR)
+				if err != nil {
+					return status.Errorf(codes.InvalidArgument, "failed to parse idevid_cert: %v", err)
+				}
+
+				// TODO: Validate this cert against Vendor CAs & check against Entity Manager.
+				// e.g., s.em.ValidateIDevID(ctx, cert, chassis)
+				session.idevidCert = cert
+				log.Infof("Successfully parsed IDevID cert for %s", cert.Subject.CommonName)
+
+				nonceBytes := make([]byte, 32)
+				if _, err := rand.Read(nonceBytes); err != nil {
+					return status.Errorf(codes.Internal, "failed to generate nonce: %v", err)
+				}
+				session.nonce = nonceBytes
+				nonceStr := base64.StdEncoding.EncodeToString(nonceBytes)
+
+				challengeResp := &bpb.BootstrapStreamResponse{
+					Type: &bpb.BootstrapStreamResponse_Challenge_{
+						Challenge: &bpb.BootstrapStreamResponse_Challenge{
+							Type: &bpb.BootstrapStreamResponse_Challenge_Nonce{
+								Nonce: nonceStr,
+							},
+						},
+					},
+				}
+				if err := stream.Send(challengeResp); err != nil {
+					return err
+				}
+				session.currentState = stateTPM20NonceSent
+				log.Infof("Sent nonce challenge to IDevID device: %s", deviceID)
+
+			case *bpb.Identity_EkPub:
+				return status.Errorf(codes.Unimplemented, "TPM 2.0 without IDevID flow not fully implemented")
+			default:
```

Comment:
Could you also include the other flows here as unimplemented? E.g. ek_ppk_pub and ppk_pub.

Reply:
That was a recent addition to the proto so you might need to pull the latest changes for that.